src/core/homepage_elements/journals/hooks.py

```diff
@@ -10,18 +10,13 @@
 
 
 def get_random_journals():
-    journals = journal_models.Journal.objects.filter(hide_from_press=False)
-    journal_pks = [journal.pk for journal in journals]
-    random_journal_pks = list()
+    journals = journal_models.Journal.objects.filter(
+        hide_from_press=False,
+    )
 
-    random.shuffle(journal_pks)
-    for i in range(0, 6):
+    sample_size = 6 if journals.count() > 6 else journals.count()
```

Comment:
Very minor: `min(6, journals.count())` to avoid the double call to `journals.count()`